std/experimental/normint.d

```diff
@@ -0,0 +1,571 @@
+// Written in the D programming language.
+
+/**
+This module implements support for normalized integers.
+
+Authors:    Manu Evans
+Copyright:  Copyright (c) 2015, Manu Evans.
+License:    $(WEB boost.org/LICENSE_1_0.txt, Boost License 1.0)
+Source:     $(PHOBOSSRC std/experimental/normint.d)
+*/
+module std.experimental.normint;
+
+import std.traits : isIntegral, isSigned, isUnsigned, isFloatingPoint;
+static import std.algorithm;
+
+import std.experimental.color : FloatTypeFor;
+
+@safe pure nothrow @nogc:
+
+
+/**
+Normalized integers express a fractional range of values within an integer data type.
+
+Unsigned integers map the values 0 -> I.max to the fractional values 0.0 -> 1.0 in equal increments.
+Signed integers represent the values -I.max -> 0 -> I.max to fractional values -1.0 -> 0.0 -> 1.0 in equal increments. I.min is outside the nominal integer range and is clamped to represent -1.0.
+
+Params: I = Any builtin integer type.
+*/
+struct NormalizedInt(I) if(isIntegral!I)
+{
+@safe:
+
+    string toString() const
+    {
+        import std.conv : to;
+        return to!string(cast(float)this);
+    }
```

Comment:
I can't work out how... Something like:
```
    void toString(scope void delegate(const(char)[]) sink) const
    {
        import std.format : formatValue, singleSpec;
        auto spec = singleSpec("%s");
        sink.formatValue(cast(double)this, spec);
    }
```
Is not `@nogc`... `singleSpec` violates.
I need a `@nogc` method of formatting the float into a string to make this work.